src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -199,26 +191,13 @@ public override Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buf
 
             WebSocketValidate.ValidateArraySegment(buffer, nameof(buffer));
 
-#if FEATURE_WASM_THREADS
-            return _innerWebSocket!.SynchronizationContext.Post(() =>
-            {
-                Task<WebSocketReceiveResult> promise;
-                lock (_thisLock)
-                {
-                    ThrowIfDisposed();
-                    promise = ReceiveAsyncCore(buffer, cancellationToken);
-                } //lock will unlock synchronously before task is resolved!
-                return promise;
-            });
-#else
-            return ReceiveAsyncCore(buffer, cancellationToken);
-#endif
+            ThrowIfDisposed();
+            return ReceiveAsyncCore(buffer, cancellationToken, fastState);
         }
 
         public override Task CloseOutputAsync(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            ThrowIfDisposed();
 
             WebSocketValidate.ValidateCloseStatus(closeStatus, statusDescription);
```

Comment:
i feel like all this state validation should occur inside the lock

Reply:
`ThrowIfDisposed` contains lock inside